### PR TITLE
Fix error handling.

### LINKED ISSUE #686: copyFileFromPod (and likely exec) does not propagate errors
Running copyFileFromPod on a non existent from file to copy results in some log lines printed but no error to the calling code:

```
java.net.ProtocolException: Expected HTTP 101 response but was '500 Internal Server Error'
    at com.squareup.okhttp.ws.WebSocketCall.createWebSocket(WebSocketCall.java:123)
    at com.squareup.okhttp.ws.WebSocketCall.access$000(WebSocketCall.java:40)
    at com.squareup.okhttp.ws.WebSocketCall$1.onResponse(WebSocketCall.java:98)
    at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:177)
    at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

If the namespace does not exist then a different log message but again not passed on for handling:

```
java.net.ProtocolException: Expected HTTP 101 response but was '404 Not Found'
    at com.squareup.okhttp.ws.WebSocketCall.createWebSocket(WebSocketCall.java:123)
    at com.squareup.okhttp.ws.WebSocketCall.access$000(WebSocketCall.java:40)
    at com.squareup.okhttp.ws.WebSocketCall$1.onResponse(WebSocketCall.java:98)
    at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:177)
    at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

There are probably other instances of this.
Exec doesn't appear to handle errors or return codes.

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #686 

cc @ceeaspb please see if this addresses your issue.

### CODE REVIEW

**Comment by yue9944882** [comment]
ci unhappy @brendandburns

**Comment by brendandburns** [comment]
CI is fixed now, please take a look.

Thanks!

File: util/src/main/java/io/kubernetes/client/Exec.java [hunk 1]
```diff
@@ -269,6 +269,22 @@ protected void handleMessage(int stream, InputStream inStream) throws IOExceptio
...
+            public void failure(Exception ex) {
+              super.failure(ex);
+              // TODO, it's possible we should suppress this error message, but currently there's
+              // no good place to surface the message, and without it, this will be really hard to
+              // debug.
+              ex.printStackTrace();
+              synchronized (ExecProcess.this) {
+                // Try for a pretty unique error code, so if someone searches they'll find this
+                // code.
+                statusCode = -1975219;
```
Comment (yue9944882) [inline util/src/main/java/io/kubernetes/client/Exec.java]: https://tools.ietf.org/html/rfc6455#section-7.4

```
   3000-3999

      Status codes in the range 3000-3999 are reserved for use by
      libraries, frameworks, and applications.  These status codes are
      registered directly with IANA.  The interpretation of these codes
      is undefined by this protocol.
```
according to the RFC, negative status code doesn't seem recommended.. how about a unique number ranged in 3000-3999?

Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/Exec.java]: This is a proces status code, not an HTTP/WebSocket status code. I think a process status code can be whatever you want it to be (and negative numbers are definitely allowed.)

Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/Exec.java]: oh tricked.. yes it's exit code from remote process.

**Pull request merged**